### PR TITLE
Delete TryGetExactPath, to remove performance penalty

### LINKED ISSUE #5836: UI performance on Version 3.00.00 RC2 is very slow
**Do you want to request a *feature* or report a *bug*?**

Report a bug

**What is the current behavior?**

General UI interaction is very slow.  

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

For example, clicking on Start > Recent Repositories takes 5 seconds, then causes the mouse to turn into a spinner for two seconds before actually displaying the recent files. Refreshing the view (F5) causes a spinner to show up for 5-6 seconds, application says Not Responding in the title bar for 2 seconds, then the view refreshes.

**What is the expected behavior?**

UI interactions for even simple Recent repositories should not take 7 seconds and cause the application to temporarily not respond.

I first uninstalled my current version, EbenZhangs GitExtensions Mod then installed 3 RC2 using the MSI. I noticed it was slow, so I tried uninstalling and reinstalled the Mod version.

To test it again, running Version 3.00.00 RC2 portable and EbenZhangs GitExtensions Mod side-by-side the differences are very noticeable.

**Environment you encounter the issue:**

- GitExtensions version:   Version 3.00.00 RC2
- GIT version: 2.17.1.windows.2
- OS version: Windows 10
- .NET version: 4.5.2

Machine specs:
Intel Core i5-7440HQ @ 2.80 GHz
32GB RAM

**Did this work in previous version of GitExtensions (which)?**

I am currently using EbenZhangs GitExtensions (Mod) 2.51

### PR DESCRIPTION
**Pull request by spdr870**
Fixes #5836

Changes proposed in this pull request:
- Remove TryGetExactPath
 
What did I do to test the code and ensure quality:
- Compile

Has been tested on (remove any that don't apply):
- GIT 2.19.1
- Windows 10

### CODE REVIEW

**Comment by amaiorano** [comment]
So with this, we no longer get the slowdown since we no longer check if paths exist. But do we handle trying to open a non-existent repo by letting the user know and asking to remove it from the recent list?

**Comment by spdr870** [comment]
@amaiorano It is still very clear that the repository doesn't exist. It shows it, but with an exclamation mark. When you click on the repository, it offers to delete it. I didn't check how it works now, but I geuss it fails to determine the current branch and this is why the exclamation mark is there.

image

**Comment by amaiorano** [comment]
That's great! Am not a member, but I approve :)

**Comment by gerhardol** [comment]
> That's great! Am not a member, but I approve :)

You can still formally review. The more reviews the better, even if more contributions increases the credibility.

**Comment by amaiorano** [comment]
> You can still formally review. The more reviews the better, even if more contributions increases the credibility.

Fair enough, I looked more closely and did ask a question about the code change.

**Comment by spdr870** [comment]
> Fair enough, I looked more closely and did ask a question about the code change.

Did you submit a code comment? I am not seeing it.

**Comment by amaiorano** [comment]
> > Fair enough, I looked more closely and did ask a question about the code change.
> 
> Did you submit a code comment? I am not seeing it.

I thought I did.. I tried to add a "Review" rather than a "Comment", and I think I didn't submit it. I'll add one now.

File: GitCommands/PathUtil.cs [hunk 1]
```diff
@@ -232,14 +232,7 @@ public static string GetDisplayPath([NotNull] string path)
...
 
-            var comparison = EnvUtils.RunningOnWindows() ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal;
-
-            if (TryGetExactPath(path, out var exactPath))
-            {
-                path = exactPath;
-            }
-
-            if (path.StartsWith(userProfile, comparison))
+            if (path.StartsWith(userProfile, StringComparison.OrdinalIgnoreCase))
```
Comment (amaiorano) [inline GitCommands/PathUtil.cs]: Any reason we don't want to check for case sensitive/insensitive based on OS? Don't we want to support Linux/Mac for the future?

Comment (RussKie) [reply GitCommands/PathUtil.cs]: Whilst not strictly necessary at this point, it would be good to keep x-plat in mind

**Comment by drewnoakes** [comment]
For reference, I added the `TryGetExactPath` feature to tidy up the list of paths that existed after I would manually type into the "open repo" textbox. They all had different capitalisation which was distracting and a bit ugly.

It's not worth the perf hit.

I think it would still be nice to keep the behaviour when _adding_ to the list of repositories. This would introduce the chance for path capitalisation to get out of date if the capitalisation is changed externally, but it would keep things looking nice in the majority case.

File: GitCommands/PathUtil.cs (hunk 1 above)
Comment (amaiorano) [reply GitCommands/PathUtil.cs]: > Even for linux, I can hardly image that the userprofile path is there with different casings.

I suppose that's true. I don't have a strong opinion on this. I suppose it's fine then.

**Comment by spdr870** [comment]
> For reference, I added the `TryGetExactPath` feature to tidy up the list of paths that existed after I would manually type into the "open repo" textbox. They all had different capitalisation which was distracting and a bit ugly.
> 
> It's not worth the perf hit.
> 
> I think it would still be nice to keep the behaviour when _adding_ to the list of repositories. This would introduce the chance for path capitalisation to get out of date if the capitalisation is changed externally, but it would keep things looking nice in the majority case.

This would be nice. But I do not have the time to do this right now. I might have some time next week to change this PR. I promised my kids to go ice-skating..:P

**Comment by spdr870** [comment]
There are multiple issues reported that will be solved with this issue. I would like to get this merged soon, without the any replacement to fix the path. I have no problems with a replacement, but during the holidays I cannot find the time to do this. @drewnoakes is this a big problem for you? I agree that having the casing repaired is nice, but investigating the side effects will take some time.

**Comment by drewnoakes** [comment]
@spdr870 could you not delete the method, and open an issue against me to look into fixing capitalisation in O(1) cases?

**Comment by spdr870** [comment]
> @spdr870 could you not delete the method, and open an issue against me to look into fixing capitalisation in O(1) cases?

Done. Forced pushed to this commit and opened issue #5963.

**Comment by spdr870** [comment]
Ok, also reverted the change in case sensitivity on other platforms...:)

**Pull request merged**